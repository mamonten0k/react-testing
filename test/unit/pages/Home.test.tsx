import React from "react";
import { it } from "@jest/globals";
import { renderRoute, renderFragment } from "../render";

import { Home } from "../../../src/client/pages/Home";

describe("<Home>", () => {
  it("Представляет собой статичный контент", () => {
    const { asFragment } = renderFragment(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("По умолчанию присвоен роут /", () => {
    const { container } = renderRoute("/");
    expect(container.firstChild).toHaveClass("Home");
  });
});
