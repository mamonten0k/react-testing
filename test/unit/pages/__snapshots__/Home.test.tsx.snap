// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> Представляет собой статичный контент 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="row"
    >
      <div
        class="col bg-secondary text-white py-4 bg-opacity-75"
      >
        <p
          class="display-3"
        >
          Welcome to Example store!
        </p>
        <p
          class="lead"
        >
          Culpa perspiciatis corporis facilis fugit similique
        </p>
        <p
          class="lead"
        >
          Cum aut ut eveniet rem cupiditate natus veritatis quia
        </p>
      </div>
    </div>
    <div
      class="row mb-4"
    >
      <div
        class="col-12 col-md-4 bg-light py-3"
      >
        <h1>
          Quickly
        </h1>
        <p
          class="lead"
        >
          Odio aut assumenda ipsam amet reprehenderit. Perspiciatis qui molestiae qui tempora quisquam
        </p>
      </div>
      <div
        class="col-12 col-md-4 bg-light py-3"
      >
        <h1>
          Qualitatively
        </h1>
        <p
          class="lead"
        >
          Ut nisi distinctio est non voluptatem. Odio aut assumenda ipsam amet reprehenderit
        </p>
      </div>
      <div
        class="col-12 col-md-4 bg-light py-3"
      >
        <h1>
          Inexpensive
        </h1>
        <p
          class="lead"
        >
          Perspiciatis qui molestiae qui tempora quisquam. Ut nisi distinctio est non voluptatem
        </p>
      </div>
    </div>
    <div
      class="row mb-4"
    >
      <div
        class="col-12py-3"
      >
        <p>
          Sed voluptatum quis voluptates laudantium incidunt laudantium. Illo non quos eos vel ipsa. Explicabo itaque est optio neque rerum provident enim qui sed. Corrupti commodi voluptatem vero soluta hic.
        </p>
        <p>
          Modi corporis consectetur aliquid sit cum tenetur enim. Sed voluptatum quis voluptates laudantium incidunt laudantium. Illo non quos eos vel ipsa. Explicabo itaque est optio neque rerum provident enim qui sed. Corrupti commodi voluptatem vero soluta hic.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
